app/forms/search_developer_form.rb

```diff
@@ -19,7 +19,7 @@ def initialize(params)
   def search
     return Developer.none if @developers_search_form.empty?
 
-    query = Developer.includes(:programming_languages).includes(:languages).includes(:developer_programmings).includes(:developer_languages)
+    query = Developer.includes(:programming_languages, :languages, :developer_programmings, :developer_languages)

```

Comment:
Do you really need to add all of them?
`includes(:programming_languages, :languages, :developer_programmings, :developer_languages)`

You should see the log, `log/development.log`, when you search against 100 developer records, for instance.
You can also ask other developers on what to do.